examples/.settings_shared/org.eclipse.jdt.core.prefs

```diff
@@ -99,23 +98,23 @@ org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentType=warni
 org.eclipse.jdt.core.compiler.problem.unlikelyCollectionMethodArgumentTypeStrict=disabled
 org.eclipse.jdt.core.compiler.problem.unlikelyEqualsArgumentType=info
 org.eclipse.jdt.core.compiler.problem.unnecessaryElse=ignore
-org.eclipse.jdt.core.compiler.problem.unnecessaryTypeCheck=ignore
+org.eclipse.jdt.core.compiler.problem.unnecessaryTypeCheck=warning
 org.eclipse.jdt.core.compiler.problem.unqualifiedFieldAccess=ignore
 org.eclipse.jdt.core.compiler.problem.unstableAutoModuleName=warning
-org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownException=ignore
-org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionExemptExceptionAndThrowable=enabled
-org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionIncludeDocCommentReference=enabled
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownException=warning
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionExemptExceptionAndThrowable=disabled
+org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionIncludeDocCommentReference=disabled
 org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionWhenOverriding=disabled
 org.eclipse.jdt.core.compiler.problem.unusedExceptionParameter=ignore
 org.eclipse.jdt.core.compiler.problem.unusedImport=error
 org.eclipse.jdt.core.compiler.problem.unusedLabel=warning
-org.eclipse.jdt.core.compiler.problem.unusedLocal=error
+org.eclipse.jdt.core.compiler.problem.unusedLocal=warning
 org.eclipse.jdt.core.compiler.problem.unusedObjectAllocation=ignore
 org.eclipse.jdt.core.compiler.problem.unusedParameter=ignore
-org.eclipse.jdt.core.compiler.problem.unusedParameterIncludeDocCommentReference=enabled
+org.eclipse.jdt.core.compiler.problem.unusedParameterIncludeDocCommentReference=disabled
```

Comment:
This one also is now more lenient. Was that intended?

If I understand this correctly (see [COMPILER_PB_UNUSED_PARAMETER_INCLUDE_DOC_COMMENT_REFERENCE](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L557:L571)), this change means that parameters mentioned as `@param` in the JavaDoc are **not** mandatory _i.e._ there could be wrong documentation (a parameter is documented but it doesn't exist). Is that correct?

Reply:
If I understand it correctly it is the other way round: If this key is set to `enabled` it means that a `@param` java-doc tag counts towards the usages of a parameter. Thus just having a parameter documented would disable an unused-warning even if the parameter is not used in the code.

I think the text in the UI also confirms this:
![grafik](https://github.com/eclipse-platform/eclipse.platform.swt/assets/44067969/8421aea6-7b4b-4751-8c9c-638190eb1200)

The settings
- [`unusedDeclaredThrownExceptionIncludeDocCommentReference`](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/6eb38a2431cb33876687312fddec8298f4ee9f6b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L952)
- [`unusedDeclaredThrownExceptionExemptExceptionAndThrowable`](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/6eb38a2431cb33876687312fddec8298f4ee9f6b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L978)

are similar.
At the other hand [`unusedDeclaredThrownExceptionWhenOverriding`](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/6eb38a2431cb33876687312fddec8298f4ee9f6b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L931) is different because setting it to `enabled` is more strict. Consequently unchecking it in the preference UI leads to the value `enabled`. I have to update that.

Reply:
> If I understand it correctly it is the other way round: If this key is set to enabled it means that a @param java-doc tag counts towards the usages of a parameter. Thus just having a parameter documented would disable an unused-warning even if the parameter is not used in the code.
>
> I think the text in the UI also confirms this:

You're right, thank you for the screenshot, it's clearer to me.

> At the other hand [unusedDeclaredThrownExceptionWhenOverriding](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/6eb38a2431cb33876687312fddec8298f4ee9f6b/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L931) is different because setting it to enabled is more strict. Consequently unchecking it in the preference UI leads to the value enabled. I have to change that.

👍